fe/fe-core/src/main/java/com/starrocks/datacache/DataCacheSelectExecutor.java

```diff
@@ -42,19 +43,27 @@ public class DataCacheSelectExecutor {
     private static final Logger LOG = LogManager.getLogger(DataCacheSelectExecutor.class);
 
     public static DataCacheSelectMetrics cacheSelect(DataCacheSelectStatement statement,
-                                                             ConnectContext connectContext) throws Exception {
+                                                     ConnectContext connectContext) throws Exception {
         InsertStmt insertStmt = statement.getInsertStmt();
 
         final WarehouseManager warehouseManager = GlobalStateMgr.getCurrentState().getWarehouseMgr();
         final Warehouse wh = warehouseManager.getWarehouse(connectContext.getCurrentWarehouseName());
+        final ComputeResourceProvider computeResourceProvider = warehouseManager.getComputeResourceProvider();
+        final List<ComputeResource> computeResources = computeResourceProvider.getComputeResources(wh);
+
         List<StmtExecutor> subStmtExecutors = Lists.newArrayList();
-        for (int workerGroupIdx = 0; workerGroupIdx < wh.getWorkerGroupIds().size(); workerGroupIdx++) {
-            long workerGroupId = wh.getWorkerGroupIds().get(workerGroupIdx);
-            ConnectContext subContext = buildCacheSelectConnectContext(statement, connectContext, workerGroupIdx == 0);
-            ComputeResource computeResource = warehouseManager.getComputeResourceProvider().ofComputeResource(
-                    wh.getId(), workerGroupId);
+        boolean isFirstSubContext = true;
+        for (ComputeResource computeResource : computeResources) {
```

Comment:
does this mean the cache select will be running on all available compute resources in the given warehouse ? @GavinMar

Reply:
> does this mean the cache select will be running on all available compute resources in the given warehouse ? @GavinMar

Yes, to avoid the cache miss after warmup caused by the same query is dispatched to another compute resource.